tests/browser_test_case.py

```diff
@@ -0,0 +1,79 @@
+import os
+import time
+
+from selenium.webdriver.chrome.options import Options
+from selenium import webdriver
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+
+from dataload.import_to_elasticsearch import import_to_elasticsearch
+
+# Test data directory
+prefix = os.path.join(os.path.dirname(__file__), "data")
+
+
+class BrowserTestCase(StaticLiveServerTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        import_to_elasticsearch(
+            [
+                os.path.join(prefix, "a002400000KeYdsAAF-currency.json"),
+                os.path.join(prefix, "a002400000OiDBQAA3.json"),
+                os.path.join(prefix, "awefw001p00000zgyHZAAY.json"),
+            ],
+            clean=True,
+            funders=os.path.join(prefix, "funders.jsonl"),
+            recipients=os.path.join(prefix, "recipients.jsonl"),
+        )
+        # elastic search needs some time to commit its data
+        time.sleep(2)
+
+    def setUp(self, *args, **kwargs):
+        BROWSER = os.environ.get("BROWSER", "ChromeHeadless")
+
+        if BROWSER == "ChromeHeadless":
+            chrome_options = Options()
+            chrome_options.add_argument("--headless")
+            # no-sandbox prevents an error when running as the root user
+            chrome_options.add_argument("--no-sandbox")
+            # uncomment this if "DevToolsActivePort" error / ubuntu snap workaround
+            chrome_options.add_argument("--remote-debugging-port=9222")
+            # chrome_options.add_argument('ignore-unexpected-deprecations')
+            self.browser = webdriver.Chrome(options=chrome_options)
+        elif BROWSER == "Firefox":
+            # Make downloads work
+            profile = webdriver.FirefoxProfile()
+            profile.set_preference("browser.download.folderList", 2)
+            profile.set_preference("browser.download.manager.showWhenStarting", False)
+            profile.set_preference("browser.download.dir", os.getcwd())
+            profile.set_preference(
+                "browser.helperApps.neverAsk.saveToDisk", "application/json"
+            )
+            self.browser = getattr(webdriver, BROWSER)(firefox_profile=profile)
+        else:
+            self.browser = getattr(webdriver, BROWSER)()
+        # Make sure we wait 5 seconds at least before slenium delcares the element no existent
+        self.browser.implicitly_wait(5)
+
+    def tearDown(self):
+        self.browser.quit()
+
+    def get(self, path):
+        """convenience function load the browser at a given path"""
+        self.browser.get(f"{self.live_server_url}{path}")
+
+    def check_js_errors(self):
+        for log in self.browser.get_log("browser"):
+            # Datatables is sending warnings that we can't currently fix
+            # https://github.com/...
```

Comment:
Is this meant to be a real link?

Reply:
Not sure, seems to be lost in time, I have raised a separate issue on this #1161 to investigate in the future.